docs/source/services.md

```diff
@@ -124,28 +124,162 @@ c.JupyterHub.services = [
 
 ## Writing your own services
 
-This Section Under Construction
+When writing your own services, you have a few decisions to make (in addition to what your service does!):
+
+1. Does my service need a public URL?
+2. Do I want JupyterHub to start/stop the service?
+3. Does my service need authentication?
+
+When a service is managed by JupyterHub,
+the Hub will pass the necessary information to the service via environment variables described above.
+To make your service most flexible, you can use these same environment variables,
+whether your service is managed by the Hub or not.
+
+When you run a service that has a url, it will be accessible under a `/services/` prefix, such as `https://myhub.horse/services/my-service/`.
+For your service to route proxied requests properly, it must take `JUPYTERHUB_SERVICE_PREFIX` into account when routing requests.
+For example, a web service would normally service its root handler at `'/'`,
+but the proxied service would need to serve `JUPYTERHUB_SERVICE_PREFIX + '/'`.
 
-```
-       /\      
-      /  \     
-     / _o \    
-    / <(\  \   
-   /   />`A \  
-  '----------'  
-```
 
 ### Authenticating with the Hub
 
-This Section Also Under Construction
+JupyterHub 0.7 introduces some utilities for using the Hub's authentication mechanism to govern access to your service.
+When a user logs into JupyterHub, the Hub sets a cookie (`jupyterhub-services`) that your service can use to authenticate requests.
+
+JupyterHub ships with a reference implementation of Hub authentication,
+but you can read on to find details of the process if you plan to implement your own hub-authenticating clients.
+
+The base implementation is the [`HubAuth`][HubAuth] class,
+which implements the requests to the Hub.
+
+To use HubAuth, you must set the `.api_token`, either when constructing the class,
+or via the `JUPYTERHUB_API_TOKEN` environment variable.
+
+Most of the implementation is in the [`HubAuth.user_for_cookie`][user_for_cookie] method,
+which makes a request of the Hub, and returns:
+
+- None if no user could be identified
+- a dict of the form:
+
+  ```python
+  {
+    "name": "username",
+    "groups": ["list", "of", "groups"],
+    "admin": False, # or True
+  }
+
+You are then free to use that user information to take appropriate action.
+
+HubAuth also caches the responses from the Hub for a number of seconds, governed by `cookie_cache_max_age` (default: five minutes).
```

Comment:
caches the Hub responses for a number of seconds, configurable by the `cookie_cache_max_age` setting (default